docs/DeveloperGuide.md

```diff
@@ -1,37 +1,293 @@
+
+**Step 2.** The `AddProjectDeadlineCommandParser` object will parse the todo input. Upon parsing the user input, it will create a `AddTodoDeadlineCommand` and return it back to `CommandHandler`.
+
+**Step 3.** After `CommandHandler` obtains the `AddTodoDeadlineCommand` returned by `AddTodoDeadlineCommandParser`, it then calls `executeCommand()` method in its own class.
+
+**Step 4.** `executeCommand()` will call `execute()` method of `AddTodoDeadlineCommand`.
+
+**Step 5.** If the number of arguments is valid, call the projectList’s `addTodoDeadline()` method, with the arguments of indexProj, indexTodo, and deadline.
+
+**Step 6.** Using the indexProj, we find the index of the project we are looking for and call its `addTodoDeadline()` method, with the arguments indexTodo and deadline.
+
+**Step 7.** Using the index of the todo, we call that specific Todo’s `setDeadline()` method and update the deadline.
+
+**Step 8.** The string deadline is used to build the class object `Deadline`
+
+**Step 9.** Inside the constructor it considers 2 types of inputs, a day of the week or a date format of yyyy-mm-dd. If it is a day of the week, it will properly detail next day that day of the week from the current day.
+
+![image info](./UmlDiagrams/AddTodoDeadline.jpg)
+
+### Proposed Features
+
+ 
+
+
+#### Tasks Due Soon Feature
+The Tasks Due Soon feature makes use of `ProjectList` and `CommandHandler` classes. It is facilitated by methods within the `Todo` and `Project` classes, stored within `ProjectList`.
+
+**Step 1.** The user executes `duesoon`. The command is retrieved and parsed by `handleUserInput()` in `CommandHandler`, which executes a self-call to `duesoon()`
+
+**Step 2.** `duesoon()` ensures that `ProjectList` has been correctly initialised before calling `printDueSoon()` in `ProjectList`
+
+**Step 3.** `printDueSoon()` in `ProjectList` initialises `tasksDue` as an empty list of `Todo`s
+
+**Step 4.** The method iterates through all `Project`s in `projectList`. For each `Project`, it then iterates through all `Todo`s in the `todos` attribute. The task deadline of each `Todo` is retrieved and compared against the current time. If the deadline is within the next 24 hours, the `Todo` is added to `tasksDue`
+
+**Step 5.** `printDueSoon()` then calls `printTodos()` in `ProjectList` and passes in `tasksDue`.
+
+**Step 6.** `printTodos()` iterates through all `Todo`s in `tasksDue` and prints each of them using `Todo`’s `toString()` method.
+
+
+
+
+
+### Previous project structure
+During v2.0, CSProjPlanner underwent a structure change to make it more OOP oriented and to make commands function in a way that are much more similar to each other.
+
+Because of this, many sequence diagrams look similar in the current version. The following sequence diagrams and explanations show what some of our features looked like at the end of v1.0:
+
+#### Add a Project
+![image info](./UmlDiagrams/addProject.png)
```

Comment:
Will it be better you enlarge the "opt: no project name provoided" to cover the commandHandler bar, as it is a part of condition check?